### PR TITLE
[lesson 11]: homework 9

### LINKED ISSUE #29: Lesson 11: homework
На базі домашки #7.

- Створити модель картки
- Створити API для створення та отримання даних картки
- База – Postgres
- Міграції обовʼязкові
- Тести для API та для методів моделі, якщо у вас будуть кастомні методи.
- API повинен бути JSON.
- Написати метод is_valid() який повертає True якщо номер валідний і False якщо ні. Метод повинен перевітяти Luhn check.

### PR DESCRIPTION
**Pull request by vmred**

### CODE REVIEW

File: lessons/lesson_11/homework_9/cards/models/card.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+class Card(models.Model):
+    id = models.AutoField(primary_key=True, editable=False)
+    pan = models.CharField(max_length=16, unique=True)
+    expiry_date = models.DateField(blank=True)
+    cvv = models.CharField()
+    issue_date = models.DateField(blank=True)
+    owner_id = models.UUIDField(default=uuid.uuid4)
+    status = models.CharField(choices=Status.choices, default=Status.NEW)
+
+    def save(self, *args, **kwargs):
```
Comment (ivan-nekipelov) [inline lessons/lesson_11/homework_9/cards/models/card.py:43]: No need to customise save method here. Just check define default values for fields. You can check code from lesson 11, there is implementation of created_date, updated_date

File: lessons/lesson_11/homework_9/cards/tests/test_unit.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import pytest
+
+from ..models.card import is_valid_card_number
```
Comment (ivan-nekipelov) [inline lessons/lesson_11/homework_9/cards/tests/test_unit.py:3]: You probably can import like from cards.models.card import is_valid_card_number

Comment (vmred) [reply lessons/lesson_11/homework_9/cards/tests/test_unit.py:3]: In such way it doesn't work, probably due to too many nested folders before app package itself

File: lessons/lesson_11/homework_9/cards/models/card.py (hunk 1 above)
Comment (vmred) [reply lessons/lesson_11/homework_9/cards/models/card.py:43]: The idea was the following:
- expiry_date and issue_date is taken from create request
- if expiry_date is null - set current date + 2 yers
- if issue_date is null - set current date

Setting like
`expiry_date = models.DateField(default=datetime.now() + timedelta(days=365 * 2))`

will raise:
```
django.db.utils.IntegrityError: null value in column "expiry_date" of relation "cards_card" violates not-null constraint
DETAIL: Failing row contains (10, 6243030000000001, null, 123, null, 00000000-0000-0000-0000-000000000001, new).
```

for create request like:
```
{
    "pan": 6243030000000001,
    "cvv": 123,
    "owner_id": 1,
    "status": "new"
}
```

**Pull request merged**